ui/components/app/simulation-details/simulation-details.stories.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import { Provider } from 'react-redux';
+import configureStore from '../../../store/store';
+import { SimulationDetails } from './simulation-details';
+import mockState from '../../../../test/data/mock-state.json';
+import { Hex } from '@metamask/utils';
+import { SimulationTokenStandard } from '@metamask/transaction-controller';
+
+const DUMMY_BALANCE_CHANGE = {
+  previousBalance: '0xIGNORED' as Hex,
+  newBalance: '0xIGNORED' as Hex,
+}
+
+const CHAIN_ID_MOCK = '0x1';
+const ERC20_TOKEN_1_MOCK = '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599'; // WBTC
+const ERC20_TOKEN_2_MOCK = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'; // USDC
+const ERC721_TOKEN_MOCK = '0x06012c8cf97bead5deae237070f9587f8e7a266d'; // CryptoKitties
+const ERC1155_TOKEN_MOCK = '0x60e4d786628fea6478f785a6d7e704777c86a7c6'; // MAYC
+
+const storeMock = configureStore({
+  metamask: {
+    ...mockState.metamask,
+    preferences: {
+      ...mockState.metamask.preferences,
+      useNativeCurrencyAsPrimaryCurrency: false,
+    },
+    providerConfig: {
+      ...mockState.metamask.providerConfig,
+      chainId: CHAIN_ID_MOCK,
+    },
+    useTokenDetection: true,
+    tokenList: {
+      [ERC20_TOKEN_1_MOCK]: {
+        address: ERC20_TOKEN_1_MOCK,
+        symbol: 'WBTC',
+        name: 'Wrapped Bitcoin',
+        iconUrl:
+          `https://static.metafi.codefi.network/api/v1/tokenIcons/1/${ERC20_TOKEN_1_MOCK}.png`,
+      },
+      [ERC20_TOKEN_2_MOCK]: {
+        address: ERC20_TOKEN_2_MOCK,
+        symbol: 'USDC',
+        name: 'USD Coin',
+        iconUrl:
+          `https://static.metafi.codefi.network/api/v1/tokenIcons/1/${ERC20_TOKEN_2_MOCK}.png`,
+      },
+      [ERC721_TOKEN_MOCK]: {
+        address: ERC721_TOKEN_MOCK,
+        symbol: 'CK',
+        name: 'CryptoKitties',
+        iconUrl:
+          `https://static.metafi.codefi.network/api/v1/tokenIcons/1/${ERC721_TOKEN_MOCK}.png`,
+      },
+      [ERC1155_TOKEN_MOCK]: {
+        address: ERC1155_TOKEN_MOCK,
+        symbol: 'MAYC',
+        name: 'Mutant Ape Yacht Club',
+        iconUrl:
+          `https://static.metafi.codefi.network/api/v1/tokenIcons/1/${ERC1155_TOKEN_MOCK}.png `
+      }
+    },
+  },
+});
+
+const meta: Meta<typeof SimulationDetails> = {
+  title: 'Components/App/SimulationDetails',
+  component: SimulationDetails,
+  decorators: [(story) => <Provider store={storeMock}>{story()}</Provider>],
+};
+
+export default meta;
+type Story = StoryObj<typeof SimulationDetails>;
+
+export const BuyErc20: Story = {
```

Comment:
Should we rename this with something more generic if it's multiple token standards?